snapshot_manager/testing_farm/watch_result.py

```diff
@@ -0,0 +1,175 @@
+    REQUEST_QUEUED = (
+        "request is queued"  # Build is waiting in queue for a backend worker.
+    )
+    REQUEST_RUNNING = (
+        "request is running"  # Backend worker is trying to acquire a builder machine.
+    )
+    TESTS_PASSED = "tests passed"  # Build in progress.
+    TESTS_FAILED = "tests failed"  # Successfully built.
+    TESTS_ERROR = "tests error"  # Build has been forked from another build.
+    TESTS_UNKNOWN = "tests unknown"  # This package was skipped, see the reason for each chroot separately.
+    PIPELINE_ERROR = "pipeline error"
+
+    def to_icon(self) -> str:
+        """Get a github markdown icon for the given testing-farm watch result.
+
+        See https://gist.github.com/rxaviers/7360908 for a list of possible icons."""
+        if self == self.REQUEST_WAITING_TO_BE_QUEUED:
+            return ":hourglass:"
+        if self == self.REQUEST_QUEUED:
+            return ":inbox_tray:"
+        if self == self.REQUEST_RUNNING:
+            return ":running:"
+        if self == self.TESTS_PASSED:
+            return ":white_check_mark:"
+        if self == self.TESTS_FAILED:
+            return ":x:"
+        if self == self.TESTS_ERROR:
+            return ":x:"
+        if self == self.TESTS_UNKNOWN:
+            return ":grey_question:"
+        if self == self.PIPELINE_ERROR:
+            return ":warning:"
+
+    @classmethod
+    def all_watch_results(cls) -> list["WatchResult"]:
+        return [s for s in WatchResult]
+
+    @property
+    def is_complete(self) -> bool:
+        """Returns True if the watch result indicates that the testing-farm
+        request has been completed.
+
+        Examples:
+
+        >>> WatchResult("tests failed").is_complete
+        True
+
+        >>> WatchResult("request is queued").is_complete
+        False
+        """
+        return self.value in [
+            self.TESTS_PASSED,
+            self.TESTS_FAILED,
+            self.TESTS_ERROR,
+        ]
+
+    @property
+    def is_error(self) -> bool:
+        """Returns True if the watch result indicates that the testing-farm
+        request has an error.
+
+        Examples:
+
+        >>> WatchResult("tests failed").is_complete
+        True
+
+        >>> WatchResult("request is queued").is_complete
+        False
+        """
+        return self.value in [
+            self.TESTS_FAILED,
+            self.TESTS_ERROR,
+        ]
+
+    @property
+    def expect_artifacts_url(self) -> bool:
+        """Returns True if for the watch result we can expect and artifacts URL in the watch output."""
+        if self.value in [
+            self.REQUEST_WAITING_TO_BE_QUEUED,
+            self.REQUEST_QUEUED,
+            self.PIPELINE_ERROR,
+        ]:
+            return False
+        return True
+
+    @classmethod
+    def is_watch_result(cls, string: str) -> bool:
+        """_summary_
+
+        Args:
+            string (str): _description_
+
+        Returns:
+            bool: _description_
```

Comment:
The usage of `_summary_` and `_description_` is coming from the old file.
I wonder if this is a good time to replace them.

Reply:
Fixed in 449cba921e034d0304dbc13f07f1c7bc4d426fb8